chalice/local.py

```diff
@@ -109,98 +181,246 @@ def create_lambda_event(self, method, path, headers, body=None):
         return event
 
 
-class ChaliceRequestHandler(BaseHTTPRequestHandler):
-
-    protocol = 'HTTP/1.1'
+class NotAuthorizedError(Exception):
+    def __init__(self, headers):
+        # type: (HeaderType) -> None
+        self.headers = headers
+
+
+class NoOptionsRouteDefined(Exception):
+    def __init__(self, headers):
+        # type: (HeaderType) -> None
+        self.headers = headers
+
+
+class LambdaContext(object):
+    def __init__(self, function_name, memory_size,
+                 runtime_millis=3000, time_source=None):
+        # type: (str, int, int, Optional[Clock]) -> None
+        if time_source is None:
+            time_source = Clock()
+        self._time_source = time_source
+        self._start_time = self._current_time_millis()
+        self._max_runtime = runtime_millis
```

Comment:
I wanted to give the unit in the name is why its not a direct mirror. I agree with `max_runtime_ms` so I'll do that.